src/int.cr

```diff
@@ -456,6 +456,19 @@ struct Int
     yield ptr, count
   end
 
+  def digits(base = 10)
+    arr = Array(typeof(self)).new(Math.log10(self.abs).ceil.to_i)
```

Comment:
Should be `(Math.log(self.abs)/Math.log(base)).ceil.to_i`

Reply:
It'd be my suspicion that calling `Math.log` and a float division would destroy any performance benefit you'd get from knowing the array size. Benchmark this with no initial capacity, an initial capacity of 20, and with `log` please.

Reply:
I did:

`digits_1`: Start with an empty array
`digits_2`: Use log and float division
`digits_3`: Start with capacity 20

Here are the results:

```
With 12312312312312312312:
digits_1   3.17M (315.65ns) (±12.16%)  448 B/op   1.32× slower
digits_2   4.13M (241.98ns) (± 6.40%)  208 B/op   1.01× slower
digits_3   4.19M (238.58ns) (± 4.44%)  208 B/op        fastest
With 123:
digits_1  13.47M ( 74.22ns) (± 6.37%)   48 B/op   1.07× slower
digits_2  14.36M ( 69.62ns) (± 7.49%)   48 B/op        fastest
digits_3   7.75M ( 129.0ns) (± 9.65%)  129 B/op   1.85× slower
```

It is at worse the same speed on both cases. Float is much much much faster than a memory allocation.

My code: https://play.crystal-lang.org/#/r/3f45

Reply:
I just added the `(Math.log(self.abs)/Math.log(base)).ceil.to_i` part because it does boost the performance. Good one, @lbguilherme! 👏🏼

The method is always returning `Array(Int32)`. Is it bad?

Reply:
Thanks for proving me wrong!